### ISSUE TITLE
Changes the rest of the document

### ISSUE THREAD

**Issue by ramontayag** [flavorjones/loofah#119]
I'm running a simple scrubber on a SOAP XML response to remove sensitive information. The problem I'm seeing is that it changes the rest of the document as well.

Loofah `2.0.3`

Script I'm running

```ruby
require 'loofah'

xml = %Q(
<x:Envelope xmlns:x="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ino="IN_OUT_API">
    <x:Header/>
    <x:Body>
        <ino:QueryListDistricts>
            <ino:USERNAME>bloom</ino:USERNAME>
            <ino:PASSWORD>abcdef</ino:PASSWORD>
            <password>123</password>
        </ino:QueryListDistricts>
    </x:Body>
</x:Envelope>
)

sensitive_info_scrubber = Loofah::Scrubber.new do |node|
  if node.name.match(/password/i)
    node.content = "[filtered]"
  end
end

Loofah.scrub_fragment(xml, sensitive_info_scrubber).to_s
```

The result:

```xml
<envelope xmlns:x="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ino="IN_OUT_API">
    <header></header>
    <body>
        <querylistdistricts>
            <username>bloom</username>
            <password>[filtered]</password>
            <password>[filtered]</password>
        </querylistdistricts>
    </body>
</envelope>
```

I expected to see the namespaces and case intact. Is this output intended?

**Comment by flavorjones** [comment]
Hi! Thanks for asking this question, and apologies for the slow reply.

If you're scrubbing an XML document, then the appropriate method to use is `scrub_xml_fragment` and not `scrub_fragment`. The former uses an XML parser which supports namespaces; the latter uses an HTML parser and so ignores namespaces (as they're not valid in HTML).

TL;DR use `scrub_xml_fragment` and your output will be

```xml
<x:Envelope xmlns:x="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ino="IN_OUT_API">
    <x:Header/>
    <x:Body>
        <ino:QueryListDistricts>
            <ino:USERNAME>bloom</ino:USERNAME>
            <ino:PASSWORD>[filtered]</ino:PASSWORD>
            <ino:password>[filtered]</ino:password>
        </ino:QueryListDistricts>
    </x:Body>
</x:Envelope>
```

**Comment by ramontayag** [comment]
Thanks for the reply nonetheless!
